File: bubblesort/bubblesort.go
package main

import (
	"fmt"
)

func bubblesort(array []int) []int {
	for i := 0; i < len(array); i++ {
		swapped := false
		for j := 0; j < len(array)-i-1; j++ {
			if array[j] > array[j+1] {
				swapped = true
				array[j], array[j+1] = array[j+1], array[j]
			}
		}
		if !(swapped) {
			break
		}
	}
	return array;
}

func main() {
	a := []int{1,2,54,2,4,1,5,2,1}
	fmt.Println(bubblesort(a))
}

File: bubblesort/bubblesort.cpp
#include <stdio.h>
#include <vector>

void bubblesort(std::vector<int>& a);

int main() {
    std::vector<int> a {1,4,2,7,1,45,1,7,2,2};
    return 0;
}

void bubblesort(std::vector<int>& a) {
    for(int i = 0; i < a.size(); i++)
    {
        bool swapped = false;
        for(int j = 0; j < a.size() - i -1; j++)
        {
            if (a[j] > a[j+1]) {
                swapped = true;
                int tmp = a[j];
                a[j] = a[j+1];
                a[j+1] = tmp;
            }
        }
        if (!swapped) break;
    }
}
